packages/nextjs/src/config/define-config.test.ts

```diff
@@ -1,27 +1,374 @@
+      defineConfigModule.defineConfig({ ...defaultConfig(), multisite: { enabled: false } });
+      const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+      expect(resultConfig.multisite?.enabled).to.be.false;
+    });
+
+    it('cookie resolution should return default function', () => {
+      defineConfigModule.defineConfig(defaultConfig());
+      const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+      expect(resultConfig.multisite?.useCookieResolution).to.be.a('function');
+
+      if (resultConfig.multisite?.useCookieResolution) {
+        process.env.VERCEL_ENV = 'preview';
+        expect(resultConfig.multisite.useCookieResolution()).to.be.true;
+        delete process.env.VERCEL_ENV;
+        expect(resultConfig.multisite.useCookieResolution()).to.be.false;
+      }
+    });
+
+    it('it should be able to override cookie resolution function', () => {
+      defineConfigModule.defineConfig({
+        ...defaultConfig(),
+        multisite: { useCookieResolution: () => true },
+      });
+      const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+      expect(resultConfig.multisite?.useCookieResolution).to.be.a('function');
+      if (resultConfig.multisite?.useCookieResolution) {
+        expect(resultConfig.multisite.useCookieResolution()).to.be.true;
+      }
+    });
+  });
+
+  describe('config.personalize', () => {
+    describe('environment variable is not set', () => {
+      it('should default to undefined', () => {
+        defineConfigModule.defineConfig(defaultConfig());
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.be.undefined;
+      });
+
+      it('should use the value from the config', () => {
+        defineConfigModule.defineConfig({
+          ...defaultConfig(),
+          personalize: { scope: 'custom-scope' },
+        });
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.equal('custom-scope');
+      });
+    });
+
+    describe('environment variable is set', () => {
+      before(() => {
+        process.env.NEXT_PUBLIC_PERSONALIZE_SCOPE = 'custom-env-scope';
+      });
+
+      after(() => {
+        delete process.env.NEXT_PUBLIC_PERSONALIZE_SCOPE;
+      });
+
+      it('should use the value from the config if present', () => {
+        defineConfigModule.defineConfig({
+          ...defaultConfig(),
+          personalize: { scope: 'custom-scope' },
+        });
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.equal('custom-scope');
+      });
+
+      it('should use the env var if config value not present', () => {
+        defineConfigModule.defineConfig(defaultConfig());
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.equal('custom-env-scope');
```

Comment:
same as above, let's group
since if new properties will appear it will be required to restructure these tests